app/models/cherrypick_task.rb

```diff
@@ -11,236 +11,46 @@
-      current_well_index = content.length
-      control_posns.select { |c| c <= current_well_index }.each do |control_well_index|
-        control_asset = control_assets[control_posns.find_index(control_well_index)]
-        add_control_request(batch, control_asset)
-      end
-      add_any_consecutive_control_requests(control_posns, batch, control_assets)
-    end
-  end
 
+  #
   # Returns a list with the destination positions for the control wells distributed by
   # using batch_id and num_plate as position generators.
-  def control_positions(batch_id, num_plate, total_wells, num_control_wells)
-    unique_number = batch_id
-
-    # Generation of the choice
-    positions = []
-    available_posns = available_control_positions(total_wells)
-
-    while positions.length < num_control_wells
-      current_size = available_posns.length
-      position = available_posns.slice!(unique_number % current_size)
-      position_for_plate = position % total_wells
-      positions.push(position_for_plate)
-      unique_number /= current_size
+  # @note wells_to_leave_free was originally hardcoded for 96 well plates at 24, in order to avoid
+  # control wells being missed in cDNA quant QC. This requirement was removed in
+  # https://github.com/sanger/sequencescape/issues/2967 however I've avoided stripping out the behaviour
+  # completely in case controls are used in other pipelines.
+  #
+  # @param batch_id [Integer] The id of the batch, used to generate a starting position
+  # @param num_plate [Integer] The plate number within the batch
+  # @param total_wells [Integer] The total number of wells on the plate
+  # @param num_control_wells [Integer] The number of control wells to lay out
+  # @param wells_to_leave_free [Integer] The number of wells to leave free at the front of the plate
+  #
+  # @return [Array<Integer>] The indexes of the control well positions
+  #
+  def control_positions(batch_id, num_plate, total_wells, num_control_wells, wells_to_leave_free: 0)
+    total_available_positions = total_wells - wells_to_leave_free
+
+    raise StandardError, 'More controls than free wells' if num_control_wells > total_available_positions
+
+    quotient = batch_id
+    size_region = (total_available_positions / num_control_wells)
+    regions = (wells_to_leave_free..(total_wells - 1)).each_slice(size_region).to_a
+
+    # Number of regions should equal number of controls, so sometimes last region is bigger than average
+    if regions.length > num_control_wells
+      last_region = regions.slice!(regions.length - 1, 1)
+      regions[regions.length - 1].concat(*last_region)
+    end
+
+    position = 0
+    regions.each_with_index.map do |region, _num_region|
+      quotient, remain = quotient.divmod(region.length)
+      # Feeding the new position with the old position seems to add more
+      # variability between the controls positions
+      position = (remain + num_plate + position) % region.length
+      region[position]
```

Comment:
Would be simpler as 
```
regions.each_with_index.map do |region, _num_region|
  remain = quotient % region.length 
  position = (remain + num_plate) % region.length
end
```

But you think changing the `quotient` each time, and adding the `position` adds something?

Reply:
Actually is the above just the same as 
```
position = (quotient + num_plate) % region.length
```